source/likely.js

```diff
@@ -2,5 +2,6 @@
 
 var likely = require('./index.js');
 
-window.likely = likely;
 window.addEventListener('load', likely.initiate);
```

Comment:
@vitkarpov, look.

Here’s how our Likely bundle looks:

```js
/* likely.js */
(function (root, factory) {
    if (typeof module === 'object' && module.exports) {
        module.exports = factory();
    } else {
        var likely = factory();
        window.addEventListener('load', likely.initiate);
        root.likely = likely;
    }
})
// ...
```

There’s also a user, and they use Webpack in their project. They do `require('likely')` somewhere in their app and then bundle the app. Here’s how their bundle looks now:

```js
/* awesome-app.js */
{
  // ...
  // One of the bundle modules
  42: (function (root, factory) {
      if (typeof module === 'object' && module.exports) {
          module.exports = factory();
      } else {
          var likely = factory();
          window.addEventListener('load', likely.initiate);
          root.likely = likely;
      }
  })
  // ...
}
```

And then the user drops their bundle into a page as a script, and the browser starts executing this:

```js
/* awesome-app.js */
{
  // ...
  // BROWSER: woah, it’s an IIFE, such a lovely IIFE, let’s execute it!
  42: (function (root, factory) {
      // BROWSER: woah, an ‘if’, I love ‘if’s! Let’s see what the condition is...
      // BROWSER: well, as I don’t have ‘module’ in my ‘window’, ‘typeof module’ doesn’t equal ‘object’,
      // BROWSER: so I’ll go to the ‘else’ branch
      if (typeof module === 'object' && module.exports) {
          module.exports = factory();
      } else {
          var likely = factory();
          // BROWSER: cool, an event listener!
          window.addEventListener('load', likely.initiate);
          root.likely = likely;
      }
  })
  // ...
}
```

You see? The `likely.js` code gets to the browser without any changes, so the browsers doesn’t find `window.module` and sets the event listener.

Of course, if a user uses `UglifyJS` plugin, it could (but I believe it wouldn’t) evaluate such `if` in `CommonJS` environment and make that code work the way we want. But this is an assumption about the user’s infrastructure, and we don’t want to make any assumptions.

Have I explained this OK? Or am I wrong in something?